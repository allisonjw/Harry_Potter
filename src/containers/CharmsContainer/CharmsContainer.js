import React from 'react';
import './CharmsContainer.scss';
import Charms from '../../components/Charms/Charms';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import '../CharmsContainer/CharmsContainer.scss';
import { PropTypes } from 'prop-types';

export const CharmsContainer = props => {
  const {allSpells } = props
 
  const displaySpells = allSpells.map(spell => {
    return (
      <Charms
      {...spell}
      key={spell._id}
      />
    );
  });

  return (
    <section className="section_charms-main">
        <section className="section_charms-heading">
          <Link to='/houses'>
            <button className='back_btn'>Back</button>
          </Link>
          <h2 className='charms_heading'> charms</h2>
          </section>
      <section className="section_charms">
        {displaySpells}
      </section>
      </section>
  )
};

export const mapStateToProps = state => ({
    allSpells: state.allSpells,
});

export default connect(mapStateToProps)(CharmsContainer);

CharmsContainer.propTypes = {
  allSpells: PropTypes.array
}
