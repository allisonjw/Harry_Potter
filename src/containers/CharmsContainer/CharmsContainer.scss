.section_charms {
    background-image: url('https://cdnb.artstation.com/p/assets/images/images/014/854/239/large/michele-di-basilio-screenshot00001.jpg?1545920711');
    background-repeat: no-repeat;
    height: 100vh;
    height: 100vh;
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(200px, 4fr));    grid-template-rows: 1fr 1fr;
    grid-template-areas: 
    "card" "card"
    "card" "card";
    overflow-x: scroll;
}
