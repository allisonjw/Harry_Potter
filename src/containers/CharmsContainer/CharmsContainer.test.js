import React from 'react';
import { shallow } from 'enzyme';
import { CharmsContainer } from '../CharmsContainer/CharmsContainer';

describe('CharmsContainer', () => {
    const mockAllSpells = [
        {
            _id: "5b74ebd5fb6fc0739646754c",
            spell: "Aberto",
            type: "Charm",
            effect: "opens objects"
        },
        {
            _id: "5b74ecfa3228320021ab622b",
            spell: "Accio",
            type: "Charm",
            effect: "Summons an object",
            __v: 0
        }
      ];

      const mockAddSpells = [
        { 
            _id: "5b74ebd5fb6fc0739646754c",
            spell: "Aberto",
            type: "Charm",
            effect: "opens objects"
        }
      ];

    const wrapper = shallow(<CharmsContainer  
        allSpells={mockAllSpells}
        addSpells={mockAddSpells}/>

);

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
});
