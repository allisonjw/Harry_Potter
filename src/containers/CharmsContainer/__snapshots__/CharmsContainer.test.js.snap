// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharmsContainer should match snapshot with all information passing in correctly 1`] = `
<section
  className="section_charms-main"
>
  <section
    className="section_charms-heading"
  >
    <Link
      to="/houses"
    >
      <button
        className="back_btn"
      >
        Back to Houses
      </button>
    </Link>
    <h2
      className="charms_heading"
    >
       charms
    </h2>
  </section>
  <p
    className="charms_matching"
  >
    Charms and Spells Matching Game
  </p>
  <section
    className="section_charms"
  >
    <Connect(Charms)
      _id="5b74ebd5fb6fc0739646754c"
      effect="opens objects"
      key="5b74ebd5fb6fc0739646754c"
      spell="Aberto"
      type="Charm"
    />
    <Connect(Charms)
      __v={0}
      _id="5b74ecfa3228320021ab622b"
      effect="Summons an object"
      key="5b74ecfa3228320021ab622b"
      spell="Accio"
      type="Charm"
    />
  </section>
</section>
`;
